### ISSUE TITLE
YC600 with 1 solar pannel : No buttons for graphics 

### ISSUE THREAD

**Issue by autourdupc** [patience4711/RPI-APS-inverters#101]
Hello.

Got an YC600 connected only with 1 solar pannel
There is no graphic button available nor statistics button.
![image](https://github.com/patience4711/RPI-APS-inverters/assets/152384092/dbea6b7d-defa-4c49-a06e-bd137138a1f7)

Got a QS1 with 4 solar pannels and got thoses buttons.
![image](https://github.com/patience4711/RPI-APS-inverters/assets/152384092/c3f219b0-bfc9-4c58-867d-1e0462f9efca)

**Comment by patience4711** [comment]
This seems impossible to me, i just tried to reproduce it with my own yc600 and get this result. This looks perfectly oke. The page displays the buttons whenever the panelcount > 0. So it seems to think that panelcount == 0.  What happened here? 
Maybe the inverterproperties file got corrupted somehow, can you examen /var/www/ecu-data/inverters/inv_properties<inverternr> should contain   a 1 for each panel like: "panels":[0, 1, 0, 0]  

If not you can edit it. Or edit the inverterpage and save it. I should be allright then.

If the properties are oke you could try to refresh the page and/or empty the browsercache.

![page_1](https://github.com/patience4711/RPI-APS-inverters/assets/12282915/d7513165-ec4c-4fce-8794-f74881c87199)

**Comment by autourdupc** [comment]
Here is the file content when opened ...
{"id":"0x2C1D","serial":40xxxxxxxxx3,"type":0,"name":"YC600-1","idx":"2","panels":[0, 1, 0, 0]}

And inverter page when opened
![image](https://github.com/patience4711/RPI-APS-inverters/assets/152384092/f5153d24-f3cc-45a8-ba65-35d86fce9c57)

After refreshing, got same result.
I also changed browser... Same issue.


Now, if I check the panel #1
![image](https://github.com/patience4711/RPI-APS-inverters/assets/152384092/6b53dbea-a0af-425a-be46-6ea13f85577d)

Then I got it !
![image](https://github.com/patience4711/RPI-APS-inverters/assets/152384092/eddb69a3-404c-4523-ae63-e056a04e8df4)


And if i remove panel #1, I lost it again.

-> Maybe there is something wrong if panel #1 not checked

**Comment by autourdupc** [comment]
Got the same issue with QS1 if I only check panel #2

![image](https://github.com/patience4711/RPI-APS-inverters/assets/152384092/b93e2790-4ec1-497d-bcb9-1a7191c30d18)


![image](https://github.com/patience4711/RPI-APS-inverters/assets/152384092/eb1e8855-65d3-46dd-b47f-85953dcef9de)

On QS1 :
If I check only #1 -> Not good
If I check only #2 -> Not good
If I check only #3 -> Not good
If I check only #4 -> Not good

All other combinations seem to work.



And finally, note also that sometimes when entering Inverter menu, the type mismatch with the name (Firefox browser). Seem to be good with Edge.

![image](https://github.com/patience4711/RPI-APS-inverters/assets/152384092/86380188-1a04-45fb-ba65-1a39242ce2dc)
Got this often !
And clicking on the menu Inv0 or Inv1 seem not working, as if some javascript / code is not loaded.

**Comment by patience4711** [comment]
This problem is very hard to solve if i cannot reproduce it. 

I have published a new software upgrade to v3_9.  In order to test, I just installed a fresh image v3_3 on a brand new zero W and upgraded to v3_9. 
Now with firefox running on a linux system, I tried  all kinds of things to fuck up with the inverterpage but it always works as expected. Always when there is at least one panel checked, no matter which one,  the buttons are displayed.
I suggest that you also upgrade to 3_9.

If you still have this problem then you can debug by editing the file /var/www/html/invpage.php.
Find the line **if(panelcount > 0)** and just before this line you add **echo "panelcount = " . $panelcount;**
Also find the line **//echo "panels". $p0a . $p1a. $p2a . $p3a;** and remove the slashes.
Save and visit/refresh the page. Now you can see which panels are checked and what the value of $panelcount is. If $panelcount > 0 the buttons should be displayed. 

If not, check the dir /var/www/ecu-data and make sure that there are files for each inverter (and no more) and check for unusual things.

As for the name mismatch, i don't know what you mean by that. This can only occur when the properties files are corrupted. The type and the name are 2 different fields, i can't see how they can be confused.

What is your hardware anyway. I installed 2 inverters on my "shadow system" inv 0 yc600 and iv 1 a qs1. If i fiddle around with the panels, no strange things happen.

**Comment by autourdupc** [comment]
I will do the tests you asked and let you know. Need for that some time. Tomorrow I'm busy. Hope to come back not too late to do some.
I updated to 3_9 and let you know.
Note that release (I mean version 3_9) still do not appear on Info 
![image](https://github.com/patience4711/RPI-APS-inverters/assets/152384092/c16b6494-1bac-4bc8-9a11-9d22d6831ba0)
![image](https://github.com/patience4711/RPI-APS-inverters/assets/152384092/965957d1-eddb-4ddf-a70a-eb1128d4040d)


Anyway, I can also give you a TeamViewer session on my computer so you can access my browser and my RPI-ECU if needed.

**Comment by autourdupc** [comment]
@patience4711 
Would like to share a video... How to MP directly to you not public ?

My hardware :
Local network with pfsense as a gateway.
NETGEAR WAC124 as Wifi access point.
 
RPI-ECU : Raspberry Pi zero 2W linked to cc2530.
Distance to inverters : QS1 : 5 meters - YC600 : 2 meters

Inverters : 
0 -> QS1 (named QS1) with 4 panels 
1 ->  YC600 (named YC600-1) with 1 panel on input #1

Got also APSYSTEM ECU-B now powered off since i use RPI-ECU.

Computers :
1 x Windows 10-64
1 x Windows 11

**Comment by patience4711** [comment]
That's another mystery, and maybe that is the whole problem. After i updated, the version 3_9 is displayed correctly. In the update script there is a line that updates the file /var/www/ecu_data/swVersion.txt. While the update is installed, all is logged so you can exactly see what is happening, you should be able to see if there are errors. So you can conduct the update again without problems and examen the output.

I have the feeling that there is something messed up with the file permissions, otherwise i couldn't explain why this doesn't work. Perhaps your unnecessary attempts to change the zimezone are the cause of this. Check if the file /var/www/ecu_data/swVersion.txt (and the whole directory) is writeable by owner and that the owner is www-data.  
Or maybe your sdcard is corrupt or your system is in a read-only state. You use an 8Gb card probably an old one?

Look, i really want to help you but this costs me too much time. To me (and many others) the software works as it should so you must be doing something wrong. 
Dont you have an old raspberry B that you can use to test with? To rule out that your hardware is the cause.

I think it's best to start over with a new sd-card.

**Comment by patience4711** [comment]
I made an image of the working version v3_9 and published that. I am going through the whole proces of downloading, burning the sd and install, just to be sure that it all works oke.
I installed it , connected to wifi and expanded the filesystem, this all worked without problems. The timezone is Europe/Amsterdam which is also correct for you so don't touch that. 

Now i registered an yc600 and a qs1 inverter, both having just one inverter. The buttons are shown...

So this is working for me, consequently it should work for you too.
I really don't know what else i could do.

**Comment by autourdupc** [comment]
@patience4711 
As I said in another post, all is working fine except panels swapped only for QS1.

I did some hack of the php pages for fun... All inverters are displayed on the same page. Each charts are auto-refreshed
And because I was not able to put the chart in the table, I used an iframe with some parameters passed to "chart_pe.php"
![image](https://github.com/patience4711/RPI-APS-inverters/assets/152384092/c7efd498-c99d-4019-8662-e389fba4038b)

This is only working on computers, phone devices need to use landscape mode to get a correct page due to size and not responsive code.

**Comment by patience4711** [comment]
Really nice. Although when you have 9 inverters....  
Since you have knowledge of php, i am still wondering why you couldn't have the buttons with only one panel. Maybe you have an idea?

**Comment by autourdupc** [comment]
Hum... I formatted the SD card and put the new image... No trace of what previously was wrong.
I notice that the html is not "clean" enought... Missing some closing instructions (like end of td or end of tr)
Also some deprecated instructions and maybe globaly some issues with styles... 
Maybe bad headers also... Not yet checked.
All of that could create unwanted results.
And also, its hard to follow the code due to php/html mixed so many times (ah ah ah !)

I also did poor html quality for testing multiple charts on the same page - It needs some time to read, check, read again, check again...
